decidim-core/app/models/decidim/scope.rb

```diff
@@ -10,6 +10,70 @@ class Scope < ApplicationRecord
                class_name: "Decidim::Organization",
                inverse_of: :scopes
 
-    validates :name, presence: true, uniqueness: { scope: :organization }
+    belongs_to  :scope_type,
+                foreign_key: "scope_type_id",
+                class_name: "Decidim::ScopeType",
+                inverse_of: :scopes
+
+    belongs_to  :parent,
+                foreign_key: "parent_id",
+                class_name: "Decidim::Scope",
+                inverse_of: :children,
+                optional: true
+
+    has_many    :children,
```

Comment:
I'd prefer having a single space between `has_many` and `children`, the same for above belongs to.